### ISSUE TITLE
TypeError: Object of type 'FieldInfo' is not JSON serializable

### ISSUE THREAD

**Issue by oxiv-dev** [samuelcolvin/pydantic#3388]
Hi! I have suddenly faced the problem that pydantic_encoder in json.py raises an error "TypeError: Object of type 'FieldInfo' is not JSON serializable". No changes were applied to code in the last two days and untill today all worked fine.
The issue was identified after a crash in Swagger UI.
Console returns the following:
```
ERROR:    Exception in ASGI application
app_1         | Traceback (most recent call last):
app_1         |   File "/usr/local/lib/python3.7/site-packages/uvicorn/protocols/http/httptools_impl.py", line 398, in run_asgi
app_1         |     result = await app(self.scope, self.receive, self.send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
app_1         |     return await self.app(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/applications.py", line 199, in __call__
app_1         |     await super().__call__(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/applications.py", line 111, in __call__
app_1         |     await self.middleware_stack(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/middleware/errors.py", line 181, in __call__
app_1         |     raise exc from None
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/middleware/errors.py", line 159, in __call__
app_1         |     await self.app(scope, receive, _send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
app_1         |     raise exc from None
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/exceptions.py", line 71, in __call__
app_1         |     await self.app(scope, receive, sender)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 566, in __call__
app_1         |     await route.handle(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 227, in handle
app_1         |     await self.app(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 41, in app
app_1         |     response = await func(request)
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/applications.py", line 152, in openapi
app_1         |     return JSONResponse(self.openapi())
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/applications.py", line 137, in openapi
app_1         |     servers=self.servers,
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/openapi/utils.py", line 355, in get_openapi
app_1         |     flat_models=flat_models, model_name_map=model_name_map
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/utils.py", line 25, in get_model_definitions
app_1         |     model, model_name_map=model_name_map, ref_prefix=REF_PREFIX
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 554, in model_process_schema
app_1         |     known_models=known_models,
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 595, in model_type_schema
app_1         |     known_models=known_models,
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 234, in field_schema
app_1         |     s, schema_overrides = get_field_info_schema(field)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 202, in get_field_info_schema
app_1         |     schema['default'] = encode_default(field.default)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 892, in encode_default
app_1         |     return t(encode_default(v) for v in dft)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 892, in <genexpr>
app_1         |     return t(encode_default(v) for v in dft)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 898, in encode_default
app_1         |     return pydantic_encoder(dft)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/json.py", line 95, in pydantic_encoder
app_1         |     raise TypeError(f"Object of type '{obj.__class__.__name__}' is not JSON serializable")
app_1         | TypeError: Object of type 'FieldInfo' is not JSON serializable
app_1         | ERROR:     call_trace=/usr/local/lib/python3.7/site-packages/uvicorn/protocols/http/httptools_impl.py - Exception in ASGI application
app_1         | Traceback (most recent call last):
app_1         |   File "/usr/local/lib/python3.7/site-packages/uvicorn/protocols/http/httptools_impl.py", line 398, in run_asgi
app_1         |     result = await app(self.scope, self.receive, self.send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/uvicorn/middleware/proxy_headers.py", line 45, in __call__
app_1         |     return await self.app(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/applications.py", line 199, in __call__
app_1         |     await super().__call__(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/applications.py", line 111, in __call__
app_1         |     await self.middleware_stack(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/middleware/errors.py", line 181, in __call__
app_1         |     raise exc from None
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/middleware/errors.py", line 159, in __call__
app_1         |     await self.app(scope, receive, _send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/exceptions.py", line 82, in __call__
app_1         |     raise exc from None
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/exceptions.py", line 71, in __call__
app_1         |     await self.app(scope, receive, sender)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 566, in __call__
app_1         |     await route.handle(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 227, in handle
app_1         |     await self.app(scope, receive, send)
app_1         |   File "/usr/local/lib/python3.7/site-packages/starlette/routing.py", line 41, in app
app_1         |     response = await func(request)
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/applications.py", line 152, in openapi
app_1         |     return JSONResponse(self.openapi())
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/applications.py", line 137, in openapi
app_1         |     servers=self.servers,
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/openapi/utils.py", line 355, in get_openapi
app_1         |     flat_models=flat_models, model_name_map=model_name_map
app_1         |   File "/usr/local/lib/python3.7/site-packages/fastapi/utils.py", line 25, in get_model_definitions
app_1         |     model, model_name_map=model_name_map, ref_prefix=REF_PREFIX
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 554, in model_process_schema
app_1         |     known_models=known_models,
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 595, in model_type_schema
app_1         |     known_models=known_models,
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 234, in field_schema
app_1         |     s, schema_overrides = get_field_info_schema(field)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 202, in get_field_info_schema
app_1         |     schema['default'] = encode_default(field.default)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 892, in encode_default
app_1         |     return t(encode_default(v) for v in dft)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 892, in <genexpr>
app_1         |     return t(encode_default(v) for v in dft)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/schema.py", line 898, in encode_default
app_1         |     return pydantic_encoder(dft)
app_1         |   File "/usr/local/lib/python3.7/site-packages/pydantic/json.py", line 95, in pydantic_encoder
app_1         |     raise TypeError(f"Object of type '{obj.__class__.__name__}' is not JSON serializable")
app_1         | TypeError: Object of type 'FieldInfo' is not JSON serializable

```
Is there some advices how to fix that?

**Comment by castrapel** [comment]
Hi @oxiv-dev , did you ever find an answer to this? I'm running into the same issue.

**Comment by DonLarry** [comment]
How did you fix that?

**Comment by KRC1997** [comment]
While developing a FastAPI application, I encountered a situation similar to the reported issue. After digging around in the code, I found that I had placed ',' characters after the attribute definitions, (given in code block below). I was able to rectify this by removing the comma characters.

`
class BriefcasePatchResponseModel(BaseModel):

    requested_action: BriefcaseActions = Field(...),
    performed_action: BriefcaseActions = Field(...),
    actioned_item: Union[BriefcaseItem, None] = Field(None)
    
    pass
`

**Comment by samuelcolvin** [comment]
(I updated your code to be more readable)

Thanks for this, so basically the field default value was a tuple.

**Comment by KRC1997** [comment]
Hey! @samuelcolvin, thanks for the update, I need to learn more about markup. :P
Yeah, it was being interpreted as a tuple

**Comment by bradleymarques** [comment]
Had the same issue - thanks

**Comment by timcroydon** [comment]
Just ran into exactly the same issue, and not for the first time - man I hate Python sometimes 😄

**Comment by metemaddar** [comment]
For me it was parentheses around Field:

```python
class BriefcasePatchResponseModel(BaseModel):
    requested_action: BriefcaseActions = (Field(...))
    performed_action: BriefcaseActions = (Field(...))
    actioned_item: Union[BriefcaseItem, None] = Field(None)
```

**Comment by codekiln** [comment]
I encountered this when creating a pydantic model intended to be used with `.with_structured_output` for use with `langchain-aws`, which doesn't support pydantic v2 yet. I created a model that extended `pydantic.v1.BaseModel`, but I was using field definitions from `pydantic.Field`. Changing these field definitions to `pydantic.v1.Field` resolved this issue. 

Note: in order for me to find which field caused it, I needed to use a debugger and set a breakpoint in `pydantic.v1.json.pydantic_encoder` on the last line.
